Fix typos found by codespell

DimitriPapadopoulos opened the pull request:
Some typos are in user-visible strings and documentation, others are in code comments.

In the long term, I recommend fixing all typos, including those in code comments, because tools like codespell do not make a difference between code comments and documentation and user-visible strings.

I have left out typos in `Contrib`, they should be fixed upstream.

DimitriPapadopoulos commented:
Not sure about `_miscrosoft_docs_are_wrong` → `_microsoft_docs_are_wrong`. I believe it's OK because it's a private variable.

sredna commented:
> I have left out typos in `Contrib`, they should be fixed upstream

We own most of Contrib. If there is anything upstream I'll deal with it.

DimitriPapadopoulos commented:
Fixed typos in `Contrib` too.

sredna commented on Include/Win/RestartManager.nsh:
```diff
@@ -34,7 +34,7 @@ RestartManager.h
 !define RmCritical 1000 ; Application is critical system process where a reboot is required to restart
 
 !define RmForceShutdown           0x1 ; Force unresponsive applications and services to shut down after the timeout period
-!define RmShutdownOnlyRegistered 0x10 ; Only shudown apps if all apps registered for restart (RegisterApplicationRestart)
```
This is actually from the Windows SDK

DimitriPapadopoulos replied:
Ah right. That would be a case where we need to fix upstream, but  the Win32 API documentation upstream appears to have been modified anyway:
[RmShutdown function (restartmanager.h)](https://docs.microsoft.com/en-us/windows/win32/api/restartmanager/nf-restartmanager-rmshutdown)
>  Shut down applications if and only if all the applications have been registered for restart using the [RegisterApplicationRestart](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-registerapplicationrestart) function

sredna replied:
I meant the typo is in the header file in the SDK, not on MSDN. I don't even think [they care](https://devblogs.microsoft.com/oldnewthing/20040705-00/?p=38573) nor do I know where to report it.

DimitriPapadopoulos commented:
I have left out `Language files` by the way.

There may be a few starting comments in English to fix, for example `appropiate` → `appropriate`:
https://github.com/kichik/nsis/blob/268379229b0c1ef00603779890a28d0345ab0240/Contrib/Language%20files/Romanian.nsh#L8

Should I fix these comments?

DimitriPapadopoulos commented:
And I just noticed part of the initial fixes, under `Docs/src/halibut`, must actually be fixed upstream.

sredna commented:
> are both `bulletted` and [`bulleted` (as in the Oxford dictionary)](https://www.oxfordlearnersdictionaries.com/definition/english/bulleted) correct in UK English?

Bulletted only has 35k hits on Google vs millions for Bulleted. I'll take the fix.

DimitriPapadopoulos commented:
I see you have merged this change into the Subversion trunk. Thanks! Would it be worth adding a configuration file `.codespellrc` to silence false positives and simplify future maintenance?

sredna commented:
I suppose that depends on how many false-positives you had to deal with and what the maintenance overhead is...

DimitriPapadopoulos commented:
I am afraid nsis has many false positives:
```sh
$ codespell | wc -l
954
$ 
```
Most of them are in translation files that should be skipped. A `.codespellrc` file could drastically reduce that number by skipping these files. For example:
```sh
$ codespell --skip 'Language files' | wc -l
226
$ 
```

The can further reduce that number by ignoring some words:
```sh
$ codespell --skip 'Language files' --ignore-words-list parms | wc -l
133
$ 
```

The benefit of a `.codespllrc` is that developers get consistent results when running `codespell` locally.

DimitriPapadopoulos committed:
Add codespell configuration file

Fix last typos in the process.

About "everytime" see for example:
https://www.grammarly.com/blog/everytime-every-time/

DimitriPapadopoulos commented:
All fixes have been merged, thank you. I have added a `.codespellrc` file and fixed the last typos. Do you think it's worth merging?

DimitriPapadopoulos commented:
Why is it not needed in this project?

I closed this PR because I am unable to update it. Following up in #33.
